_sources/contributing.rst.txt

```diff
@@ -421,18 +475,52 @@ When writing docstrings, follow the `NumPy template <https://numpydoc.readthedoc
       wikipedia page.
       """
 
+And here's a filled-in example from the Scikit-Learn project, modified to our syntax::
+
+   def fit_predict(self, X, y=None, sample_weight=None):
+      """Compute cluster centers and predict cluster index for each sample.
+
+      Convenience method; equivalent to calling fit(X) followed by
+      predict(X).
+
+      Parameters
+      ----------
+      X : {array-like, sparse_matrix} of shape=[n_samples, n_features]
+         New data to transform.
+      y : Ignored
+         Not used, present here for API consistency by convention.
+      sample_weight : array-like, shape [n_samples,], optional
+         The weights for each observation in X. If None, all observations
+         are assigned equal weight (default: None).
+
+      Returns
+      -------
+      labels : array, shape=[n_samples,]
+         Index of the cluster each sample belongs to.
+      """
+      return self.fit(X, sample_weight=sample_weight).labels_
+
 In general, have the following in mind:
+ 
    1. Use Python basic types. (``bool`` instead of ``boolean``)
+ 
    2. Use ``[`` for defining shapes: ``array-like, shape=[n_samples,]``
+ 
    3. For strings with multiple options, use brackets:
       ``input: str, {'log', 'squared', 'multinomial'}``
+ 
    4. 1D or 2D data can be a subset of
       ``{array-like, ndarray, sparse matrix, dataframe}``. Note that ``array-like``
       can also be a ``list``, while ``ndarray`` is explicitly only a ``numpy.ndarray``.
+ 
```

Comment:
Maybe also add the comments from our discussion, to have it 100% clear?

- Description and parameters' description starts with uppercase, ends with a .

- optional is on the same line as the parameters' type.

- There are spaces around :

- The description of a method starts with an imperative verb, OK to use Compute if needed (but not, Calculate, Evaluate, etc, let's agree on Compute).

Reply:
- Description beginning with uppercase and ending with period is in the current version. I'll add it to the Parameters section's description as well.

- Thanks, I'll add this (optional info)

- Thanks, I'll add this (`` :  ``)

- I don't think the Description section has to start with an imperative verb, only the Summary section. Also, this point (imperative verb) is written in the Summary section,  but I'll add the compute vs calculate point next. :)